new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSystemResOverrides.java

```diff
@@ -0,0 +1,514 @@
+      image = kindRepoImage;
+      isUseSecret = false;
+    } else {
+      // create pull secrets for WebLogic image when running in non Kind Kubernetes cluster
+      createOCRRepoSecret(domainNamespace);
+    }
+    //create and start WebLogic domain
+    createDomain();
+
+    // build the sitconfig application
+    Path distDir = buildApplication(Paths.get(APP_DIR, "sitconfig"),
+        null, null, "dist", domainNamespace);
+    sitconfigAppPath = Paths.get(distDir.toString(), "sitconfig.war");
+    assertTrue(sitconfigAppPath.toFile().exists(), "Application archive is not available");
+
+    //deploy application to view server configuration
+    deployApplication(clusterName + "," + adminServerName);
+
+  }
+
+  /**
+   * Test JMS and WLDF system resources configurations are overridden dynamically when domain resource
+   * is updated with overridesConfigMap property. After the override verifies the system resources properties
+   * are overridden as per the values set.
+   */
+  @Test
+  @DisplayName("Test JMS and WLDF system resources override")
+  public void testSystemResourceOverride() {
+
+    //store the pod creation timestamps
+    storePodCreationTimestamps();
+
+    List<Path> overrideFiles = new ArrayList<>();
+    overrideFiles.add(
+        Paths.get(RESOURCE_DIR, "configfiles/configoverridesset2/jms-ClusterJmsSystemResource.xml"));
+    overrideFiles.add(Paths.get(RESOURCE_DIR, "configfiles/configoverridesset2/diagnostics-WLDF-MODULE-0.xml"));
+    overrideFiles.add(Paths.get(RESOURCE_DIR, "configfiles/configoverridesset2/version.txt"));
+
+    //create config override map
+    createConfigMapFromFiles(overridecm, overrideFiles, domainNamespace);
+
+    String introspectVersion = assertDoesNotThrow(() -> getNextIntrospectVersion(domainUid, domainNamespace));
+
+    logger.info("patch the domain resource with overridesConfigMap and introspectVersion");
+    String patchStr
+        = "["
+        + "{\"op\": \"add\", \"path\": \"/spec/configuration/overridesConfigMap\", \"value\": \"" + overridecm + "\"},"
+        + "{\"op\": \"add\", \"path\": \"/spec/introspectVersion\", \"value\": \"" + introspectVersion + "\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    verifyIntrospectorRuns();
+    verifyPodsStateNotChanged();
+
+    //wait until config is updated upto 5 minutes
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for jms server configuration to be updated"
+                + "(elapsed time {0} ms, remaining time {1} ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(configUpdated());
+
+    verifyJMSResourceOverride();
```

Comment:
Can we make  REST call to Admin Server to get System Resource Runtime MBean to verify the Override, instead of using a web application to check the override ?

Reply:
It will be too much in the test code. For JMS its ok, but in WLDF there are lot of overrides that needs to be verified so it is better to stick with the application.

Reply:
OK. It is fine